classes/class-p4-custom-taxonomy.php

```diff
@@ -235,12 +260,12 @@ public function save_taxonomy_page_type( $post_id, $post ) {
 				$terms = wp_get_object_terms( $post_id, self::TAXONOMY );
 				if ( ! is_wp_error( $terms ) ) {
 
-					$all_terms = $this->get_terms();
+					$default_p4_pagetype = $this->get_default_p4_pagetype();
 
-					// Assign taxonomy's first term, if no term is assigned to post.
+					// Assign default p4-pagetype, if no term is assigned to post.
 					if ( empty( $terms ) ) {
-						if ( ! is_wp_error( $all_terms ) && ! empty( $all_terms ) && is_object( $all_terms[0] ) ) {
-							wp_set_post_terms( $post_id, $all_terms[0]->slug, self::TAXONOMY );
+						if ( ! is_wp_error( $all_terms ) && ! empty( $all_terms ) && is_object( $default_p4_pagetype ) ) {
```

Comment:
@sagarsdeshmukh Since you have introduced get_default_p4_pagetype function the checks for $all_terms variable are not needed.

Reply:
Good catch :+1:  My mistake I will update it.